contracts/profit/src/state/mod.rs

```diff
@@ -36,11 +38,52 @@ const STATE: Item<'static, State> = Item::new("contract_state");
 
 type IcaConnector = dex::IcaConnector<OpenIca, ContractResult<DexResponse<Idle>>>;
 
+pub(crate) struct ConfigAndResponse {
+    pub config: Config,
+    pub response: PlatformResponse,
+}
+
+pub(crate) struct StateAndResponse<T> {
+    pub state: T,
+    pub response: PlatformResponse,
+}
+
+impl<T> StateAndResponse<T> {
+    pub fn map_state<U>(self) -> StateAndResponse<U>
+    where
+        T: Into<U>,
+    {
+        StateAndResponse {
+            state: self.state.into(),
+            response: self.response,
+        }
+    }
+}
+
```

Comment:
let's not reinvent the wheel, there is `platform::state_machine::Response` which carries both messages and state, and the state includes the new config

in addition, let's not complicate the code with `WithConfig` trait is not needed

Reply:
As I mentioned in a comment before, I wanted to avoid repetition and this trait was the way to do it. If you insist, I will remote it but that will just reintroduce duplication.